From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Paalanen <pekka.paalanen@collabora.com>
Date: Tue, 10 Jun 2025 14:58:26 +0300
Subject: staging/color-management: recommend gamma22 instead of srgb

https://gitlab.freedesktop.org/pq/color-and-hdr/-/blob/main/doc/wayland_qa.md#q-should-srgb-content-be-decoded-with-the-piecewise-srgb-transfer-function

I found it unfortunate that we left in a TF code that will intuitively
be used incorrectly. It is as if we designed the protocol so that
compositors will need to fix up client image descriptions.

I am not aware of any use case that would want to target a display with
the sRGB piece-wise transfer function, that would be a non-standard
display.

This patch does not help compositors avoid needing to second-guess
client image descriptions using srgb TF, but it at least documents the
situation. We could choose to out-law srgb TF in a minor version bump,
or drop it completely in the next major version. Compositors can also
not advertise support for srgb TF.

Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>

diff --git a/staging/color-management/color-management-v1.xml b/staging/color-management/color-management-v1.xml
index ee482c5af1ae0cd78219f1f64baab87840e32a86..4c1bc759c390fc231e9127461fe22aeb2333d00e 100644
--- a/staging/color-management/color-management-v1.xml
+++ b/staging/color-management/color-management-v1.xml
@@ -290,6 +290,8 @@
             of Federal Regulations 73.682 (a) (20)
           - Rec. ITU-R BT.1700-0 625 PAL and 625 SECAM
           Equivalent to H.273 TransferCharacteristics code point 4.
+
+          Note: an sRGB display (IEC 61966-2-1) uses this transfer function.
         </description>
       </entry>
       <entry name="gamma28" value="3">
@@ -340,6 +342,10 @@
           - IEC 61966-2-1 sRGB
           Equivalent to H.273 TransferCharacteristics code point 13 with
           MatrixCoefficients set to 0.
+
+          Note: This is not appropriate for describing sRGB material.
+          sRGB material is intended to be viewed on an sRGB display, and
+          that is described by gamma22.
         </description>
       </entry>
       <entry name="ext_srgb" value="10">
